Improve Filter Editor UI

cjcenizal opened the pull request:
Based on https://github.com/elastic/kibana/pull/11375#pullrequestreview-41411144 

## Changes

* Use labels consistently in Filter Editor dropdown.
* Improve screen reader and keyboard accessibility.
* Convert 'Edit Query DSL' button to a checkbox.

CC @snide 

![filter_editor_checkbox](https://cloud.githubusercontent.com/assets/1238659/26764053/2a6ad31a-4914-11e7-9061-69c3a6cbafd3.gif)

snide commented:
I'd go for a straight toggle link rather than the checkbox.

cjcenizal commented:
Good idea @snide! Here's how it looks now.

![image](https://cloud.githubusercontent.com/assets/1238659/26765649/298989e8-4935-11e7-934c-74cd8956cfcb.png)

![image](https://cloud.githubusercontent.com/assets/1238659/26765651/2b0b82a8-4935-11e7-9e2b-d43e564d2c87.png)

@Bargs and @lukasolson I changed the language from "Edit query DSL" to "Edit raw query" because I don't think most people know what a domain-specific language is. Thoughts?

cjcenizal commented on src/ui/public/filter_editor/filter_editor.html:
```diff
@@ -6,110 +6,115 @@
+      <!-- Filter dropdowns -->
+      <div
+        class="kuiFieldGroup kuiVerticalRhythmSmall kuiFieldGroup--alignTop"
+        ng-if="!filterEditor.showQueryDslEditor()"
```
Was there any reason we chose to not use `ng-if` here originally? Will this disrupt anything?

cjcenizal replied:
Answered my own question. `ng-if` adds an isolate scope which makes it difficult to access controller properties.

cjcenizal commented on src/ui/public/filter_editor/filter_editor.html:
```diff
@@ -6,110 +6,115 @@
+      <!-- DSL editor -->
+      <div
+        class="kuiVerticalRhythmSmall"
+        ng-if="filterEditor.showQueryDslEditor()"
```
Same question re using an `ng-if`.

Bargs commented:
> I changed the language from "Edit query DSL" to "Edit raw query" because I don't think most people know what a domain-specific language is. Thoughts?

The one benefit of calling it query DSL that I can think of is that it distinguishes it from other possible types of "raw queries", like a lucene query string. When creating a filter from scratch the editor only contains `{}` so mentioning the query dsl may provide a helpful hint about what's valid input.

cjcenizal commented:
@Bargs Good point. Is https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html the best reference we have available for explaining the Query DSL? And am I correct in understanding that this use of the Query DSL involves the [filter context](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-filter-context.html)?

Bargs commented:
> Is https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html the best reference we have available for explaining the Query DSL?

Yeah that seems good to me.

> And am I correct in understanding that this use of the Query DSL involves the filter context?

Ironically, our filters do not currently go under the filter context.

cjcenizal commented:
@Bargs I changed the language back, and added a link to the docs. Thoughts?

![image](https://cloud.githubusercontent.com/assets/1238659/26791683/b3e1a0de-49cc-11e7-9053-373d036f3c82.png)

![image](https://cloud.githubusercontent.com/assets/1238659/26791687/b731fbb2-49cc-11e7-8e6b-a4229af8b873.png)

Bargs commented:
I like the addition of the doc link. Could we just have the text say "Filters are built using the Elasticsearch Query DSL."? Query vs filter context doesn't change how the user writes the DSL, so it might just cause unnecessary confusion. I'd also sort of consider it an implementation detail of Kibana that could change.
